Create HLTONLY*EventContent

Dr15Jones opened the pull request:
#### PR description:

This can be used with the two file solution to allow the HLT only step to not write out the RAW data which improves performance. The following step can then read this output and the original RAW file.

#### PR validation:

The 139.00x RelVals appear to be the only ones that have an HLT only step. I changed their step 2 to only use the HLTONLY content and told step3 to use `--secondfilein` to read the RAW file as well as step2.root. The job ran fine.

Dr15Jones commented:
The full set of what I did was

1. runTheMatrix.py -l 139.003
2. ln -s 139.003_RunHLTPhy2021+RunHLTPhy2021+HLTDR3_2021+RECODR3_HLTPhysics_Offline+HARVESTD2021HLTPhy/step1_* .
3. cmsDriver.py cmsDriver.py step2 --process reHLT -s L1REPACK:Full,HLT:@relval2021 --conditions auto:run3_hlt_relval --data --eventcontent HLTONLY --datatier HLTONLY --era Run3 -n 100 --filein filelist:step1_dasquery.log --lumiToProcess step1_lumiRanges.log --fileout file:step2.root --nThreads 4
4. cmsDriver.py cmsDriver.py step3 --conditions auto:run3_data_relval -s RAW2DIGI,L1Reco,RECO,PAT,ALCA:TkAlMinBias+HcalCalIterativePhiSym+HcalCalIsoTrkProducerFilter+HcalCalHO+HcalCalHBHEMuonProducerFilter,DQM:@commonReduced+@miniAODDQM --datatier RECO,MINIAOD,DQMIO --eventcontent RECO,MINIAOD,DQM --data --process reRECO --scenario pp --era Run3 --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run3 --procModifiers siPixelQualityRawToDigi -n 100 --filein file:step2.root --secondfilein root://cmsxrootd-site.fnal.gov//store/data/Commissioning2021/HLTPhysics/RAW/v1/000/346/512/00000/9e1384d9-83b8-4240-8486-950fa0e22a77.root --fileout file:step3.root --nThreads 4

Dr15Jones commented:
@makortel FYI

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37560/29291

- This PR adds an extra 28KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLTrigger_EventContent_cff.py  modified in PR(s): #37551

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

- Configuration/EventContent (**operations**)
- HLTrigger/Configuration (**hlt**)
- L1Trigger/Configuration (**l1**)


@perrotta, @Martin-Grunewald, @rekovic, @epalencia, @cecilecaillol, @cmsbuild, @missirol, @qliphy, @fabiocos, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @missirol, @silviodonato, @fabiocos this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Dr15Jones commented:
Please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c8a30f/23892/summary.html
**COMMIT**: f2f3f69c6c8657e4dabc869c43811565d9de4a54
**CMSSW**: CMSSW_12_4_X_2022-04-13-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37560/23892/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3593039
 - DQMHistoTests: Total failures: 8
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3593009
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 200 log files, 45 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_4_X_2022-04-13-1100+c8a30f/49590/triggerResults): no differences found

Martin-Grunewald commented:
This will not work: the file ```HLTrigger/Configuration/python/HLTrigger_EventContent_cff.py``` is generated from ConfDb and thus overwritten by any changes touching it there. The changes to the file in this PR would thus be overwritten and lost.

Martin-Grunewald commented:
It looks like the new ```HLTriggerONLY  = cms.PSet``` is identical to the already existing ```HLTriggerAOD  = cms.PSet``` so one could simply use that!

Dr15Jones commented:
@Martin-Grunewald Thanks for looking at this.

> It looks like the new` HLTriggerONLY = cms.PSet` is identical to the already existing `HLTriggerAOD = cms.PSe`t and `HLTriggerRECO = cms.PSet` (these two are also equal) so one could simply use one of those!

Just in case `HLTriggerAOD` ever changed, I'd feel better to not couple them to this change. I'm taking a different tact which will not change of the L1 or HLT event content declarations.

Dr15Jones committed:
Create HLTONLY*EventContent

This can be used with the two file solution to allow the HLT only
step to not write out the RAW data which improves performance. The
following step can then read this output and the original RAW file.

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37560/29317

- This PR adds an extra 16KB to repository

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c8a30f/23923/summary.html
**COMMIT**: 2496e039495b45c16be07bcb0f5a4f23ea8b53c1
**CMSSW**: CMSSW_12_4_X_2022-04-14-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37560/23923/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3593039
 - DQMHistoTests: Total failures: 13
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3593003
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 47 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 200 log files, 45 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_4_X_2022-04-14-1100+c8a30f/49608/triggerResults): no differences found

qliphy commented:
please test
to refresh

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c8a30f/24171/summary.html
**COMMIT**: 2496e039495b45c16be07bcb0f5a4f23ea8b53c1
**CMSSW**: CMSSW_12_4_X_2022-04-24-0000/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37560/24171/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /pool/condor/dir_8329/jenkins/workspace/compare-root-files-short-matrix/data/PR-c8a30f/39434.75_TTbar_14TeV+2026D88_HLT75e33+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HLT75e33+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3695434
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3695404
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 48 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 205 log files, 45 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_4_X_2022-04-24-0000+c8a30f/49812/triggerResults): no differences found

perrotta commented:
assign hlt

Martin-Grunewald commented:
Chris and I discussed this already and he adapted the way the event content is built, to be fine from our side.

Martin-Grunewald commented:
+1

perrotta commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will be automatically merged.
